### ISSUE TITLE
Password strength calculation seems inconsistent

### ISSUE THREAD

**Issue by mennowame** [rollerworks/PasswordStrengthBundle#46]
The password strength calculation seems to be inconsistent with the documentation and itself. There are big differences in scores that you wouldn't expect (based on documentation), for example:
abcd! will have strength 0
1234! will have strength 2

There are a few issues:

* alpha characters don't get a strength point, while digits and special chars do get a point
* combination of alpha and special char will get a minus point, combination of digit and special char wont get a minus point
* only alpha or only digit will get a minus point, only special chars wont get a minus point
* minimum password length is a requirement that is checked outside of strength, so in my opinion doesn't need to contribute to strength. password length can now give 2 points (minimum length and 12 char length). I would suggest to only give a point for 12 char length.

Based on this I would like to suggest a change to the password strength calculation. Feel free to tell me I'm wrong :wink: 
Basically there are are a few criteria that can make a password safe
* alpha character
* digit
* special character
* both lowercase and uppercase alpha character
* long password length (>12)

If you give a point when passing one of these criteria you get a minimum of 1 and a maximum of 5. No minus points are needed, and minimum length will already be enforced with the password length check.

I've already made a version that includes the logic above, and that seems to work fine. Before making a pull request I've still got to fix all tests. I hope to finish that this week.

**Comment by sstok** [comment]
:+1: The PasswordStrengthValidator was originally based on a JavaScript version and has some issues indeed so any contribution to improve this more then welcome.

**Comment by mennowame** [comment]
Ok I'll finish the tests and make a pull new pull request
